source/config/Config.cpp

```diff
@@ -287,7 +287,7 @@ bool PlainConfig::LoadFromCliArgs(const CliArgs &cliArgs)
     return loadFeatureCliArgs;
 }
 
-bool PlainConfig::LoadFromEnvironment()
+void PlainConfig::LoadMemTraceLevelFromEnvironment()
```

Comment:
@HarshGandhi-AWS This solution will work, but what if we completely remove the initialization order dependency between Config and SharedCrtResourceManager by doing the following 2 steps? 

1. Copy-paste the code from `PlainConfig::LoadMemTraceLevelFromEnvironment()` to `SharedCrtResourceManager::initializeAllocator`.
2. Invoke `SharedCrtResourceManager::initializeAllocator`  from the `SharedCrtResourceManager` constructor. 

This has the following benefits:
* Config does not have to be initialized before SharedCrtResourceManager.
* Unit tests that use SharedCrtResourceManager do not require a fixture to explicitly initialize SharedCrtResourceManager for memory tracing.
* main.cpp requires only a one-line change to invoke the `resourceManager = std::make_shared<SharedCrtResourceManager>();`

Reply:
Calling the initializeAllocator() method from SharedCrtResourceManager constructor was causing some Sensor Publish feature unit test failures. To avoid rewriting the unit tests, I have removed initializeAllocator() method call from the constructor and I am manually calling the initializeAllocator() method from main and some unit tests wherever the Cjson library was used.